### PR TITLE
[Genesis] Update the non-test genesis flow to destroy aptos framework account's mint capability

### PR DESCRIPTION
**Pull request by movekevin**
### Description
The aptos framework account needs to temporarily have mint capability of Aptos Coin so genesis can mint coins to initial validators. Once genesis is complete, we need to destroy the mint capability to ensure not even the aptos framework account (which the on chain governance can control) can mint Aptos coins later.

### Test Plan
New e2e test.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2917)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 1]
```diff
@@ -149,12 +149,17 @@ module aptos_framework::genesis {
...
             stake::add_stake(&owner_account, amount);
             stake::join_validator_set_internal(&owner_account, *owner);
 
             i = i + 1;
         };
+
+        // Destroy the aptos framework account's ability to mint coins now that we're done with setting up the initial
+        // validators.
+        aptos_coin::destroy_mint_cap(aptos_framework);
+
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/genesis.move:162]: we shouldn't do this, we should make this either opt-in or opt-out -- that way testnets can continue to have unlimited faucets.

**Review by davidiw — CHANGES_REQUESTED** [review]
I am concerned that the existing faucet didn't fail this... do we not have tests to validate that new accounts are created and given a copy of the mint capability?

we may need to discuss this more.

**Comment by davidiw** [comment]
@gregnazario , @CapCap

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:162]: The faucet in tests/testnets use the core resources account, which already has mint capability and can be controlled via the is_test flag in genesis configurations

**Comment by movekevin** [comment]
> I am concerned that the existing faucet didn't fail this... do we not have tests to validate that new accounts are created and given a copy of the mint capability?
> 
> we may need to discuss this more.

This issue is only for the flow where is_test = true. We currently set is_test = false for all tests/devnet/testnets.

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 1 above)
Comment (gregnazario) [reply aptos-move/framework/aptos-framework/sources/genesis.move:175]: He added the `is_test` flag to allow this to be controlled.

Though why is it destroyed here?  I don't see the configuration for `is_test`

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:175]: We need this removal for both tests/testnets and mainnet. For testnets, the core resources account already has the mint capability and will keep it. For both testnets and mainnet, aptos framework account only needs temporary mint power to mint to initial validators and accounts. Once it's done, it no longer needs mint cap.

**Review by davidiw — APPROVED** [review]
I mean this is fine, but I think for mainnet, we should consider a script that creates accounts and stores funds on them independent of this flows.

**Pull request merged**